### ISSUE TITLE
Merge develop branch into main branch

### ISSUE THREAD

**Issue by LoTerence** [hackforla/CivicTechJobs#610]
### Overview

As a developer, we need the `develop` and `main` branches to be in sync so that the code base is clean and we can more easily collaborate.

Deadline: 11/14/2024

### Action Items

- [x] Create a protected branch of main called `ava-main-v1` to preserve all their old work. This will separate v1 from v2 (pre backend reset vs post backend reset).
- [x] Merge `main` into `develop`
- [x] Test all the changes in `develop`. Ideally in a staging environment
- [x] Review code (175 files changed)
- [x] Double check github actions to see if any of them are effected.
- [x] Mkdocs gh action needs to be updated to deploy from main
- [x] Execute pull request to Merge `develop` into `main`

This issue Depends on these deployment tasks:
- [x] Important: deploy `develop` branch to (personal) AWS as a staging environment to fully test current changes.
- [x] we need an understanding of the exact process of how the `main` branch is currently deployed to AWS incubator
- [x] Integrate whitenoise
- [x] prepare `develop` branch for a staging deployment

The deployment for the `develop` branch should be totally separate from what we have now ([stage.civictechjobs.org](https://stage.civictechjobs.org/)). This way we can test and confirm a `develop` deployment is fully functional before pushing to main.

Deprioritized tasks:
- [x] Write documentation about new `main` branch and old main branch - moved to issue #598 
- [x] Write Documentation about how the `stage` docker environment works. (issue 598)
- [x] update documentation about how mkdocs is deployed - it is deployed from `main` now

New issue found while testing stage environment:
- [x] qualifier page failing to load due to lazy loading not working in current project set up.

### Resources/Instructions

https://stage.civictechjobs.org/

[Resources](https://hackforla.github.io/CivicTechJobs/resources/)

gh action v4 migration docs:
- https://github.com/actions/upload-artifact?tab=readme-ov-file#breaking-changes
- https://github.com/actions/upload-artifact/blob/main/docs/MIGRATION.md

Deploying a django app:
- https://developer.mozilla.org/en-US/docs/Learn/Server-side/Django/Deployment
- https://docs.djangoproject.com/en/5.1/howto/deployment/

**Comment by LoTerence** [comment]
Update 11/21/2024

- The `develop` branch has been successfully deployed to our personal AWS account for testing.
- Incubator will not be able to help us deploy until January.

This issue will no longer depend on an incubator depoyment. I will test staging on my personal AWS, fix issues found there, then merge to `main`. 

It should be okay because the gh action to deploy the `main` branch to stage.civictechjobs.org is non-functional and disabled.

Our goal in January will be to work with the incubator/devops team when they are available to fully complete the deployment pipelines. We will sync the new `main` with a prod deployment, and `develop` to stage.civictechjobs.org then.

**Comment by LoTerence** [comment]
This issue depends on #585
